iqama_fetcher.py

```diff
@@ -1,42 +1,114 @@
-"""Fetches iqama times from our heroku server."""
+"""Fetches iqamah times from the Iqamah.net service."""
 
-import json
+import re
+from xml.etree import ElementTree
 import util
 import requests
+from masjid_util import GetIqamaID
 
-_SCRAPER_URL = 'https://islam-buddy-staging.herokuapp.com/iqama'
-
+_IQAMAH_NET_URL = 'http://feed.iqamah.net/IQ'
 
 def GetIqamaTime(desired_prayer, masjid):
-  """Gets the iqama time from the heruko scraper service by peforming a POST.
+  """Gets the iqama time from the iqamah.net service by peforming a GET.
 
   Args:
     desired_prayer: DailyPrayer enum representing desired salah
     masjid: a string representing the masjid of intereset
 
-  Returns: a dict containing the prayers and iqama times
+  Returns: a string containing the iqama time along with AM/PM
   """
 
-  # set up the parameters in the format expected by the scraper service
-  post_params = {
-      'masjid': masjid,
-  }
-  print 'scraper get_params = ', post_params
-  request = requests.post(_SCRAPER_URL, params=post_params, timeout=15)
+  # making sure the masjid input is in UTF-8 format
+  masjid_encoded = util.EncodeParameter(masjid, 0)
+  # getting the id from the _MASJID_METADATA
+  masjid_id = GetIqamaID(masjid_encoded)
+
+  desired_prayer = util.GetPrayerKeyName(desired_prayer)
+
+  print 'Iqamah.net Masjid ID ', masjid_id
+
+  # if masjid_id doesn't exist then return None
+  if not masjid_id:
+    return None
+
+  request = requests.get(_IQAMAH_NET_URL + str(masjid_id) + ".xml", timeout=15)
 
   # checks if request isn't bad
   # pylint: disable=no-member
   if request.status_code == requests.codes.ok:
-    response = json.loads(request.text)
-    print 'response from scraper service = ', response
-
-    # translate the dict keys to enums
-    iqama_times = {}
-    for key in response:
-      prayer = util.StringToDailyPrayer(key)
-      if prayer:
-        iqama_times[prayer] = response[key]
-
-    return iqama_times[desired_prayer]
-  print 'bad response from scraper service = ', request.status_code
+    # setting a tree node to the beginning of request xml content
+    tree = ElementTree.fromstring(request.content)
+    # setting an element node to element 'IqDay' which contains the
+    # prayer names and timings
+    # <IqDay><Fajr>##:##</Fajr>.....</IqDay>
+    iqamah_element = tree.findall('IqDay')[0]
+
+    # looping through the IqDay tags (prayer names)
+    for index in range(0, len(iqamah_element)):
+      # convert the prayer names in the IqDay element to a known format
+      iqamah_name = GetEquivalentPrayer(str(iqamah_element[index].tag))
```

Comment:
i dont think so since it pulls both prayer names and times i assume
also this is fail safe in case the guy decides to change the names for whatever reason